replica/database.hh

```diff
@@ -581,6 +581,9 @@ private:
     std::pair<size_t, locator::tablet_range_side> storage_group_of(dht::token token) const noexcept;
     size_t storage_group_id_for_token(dht::token token) const noexcept;
     storage_group* storage_group_for_token(dht::token token) const noexcept;
+    // Returns a pointer to the storage_group at the given index
+    // if it's owned by this shard, otherwise return nullptr.
+    storage_group* storage_group_at(size_t idx) const noexcept;
```

Comment:
Name should reflect the special case, e.g.: `shard_local_storage_group_at()`.
Does `storage_group_for_token()` work the same way? In that case, the current name is more consistent.

Reply:
storage_group_for_token is called on the write path.  With streaming it inserts sstables into the storage_group while ownership of the tablet replica is still pending so storage_group_at would still return null during migration, until it's done.

Reply:
I'll rename to `shard_local_storage_group_at` to reflect the semantics.